File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        
        bool removed = false;
        for(int i=0;i<s.size();i++)
        {
            if(s[i] == 'A')
            {
                if(i+1 < s.size())
                {
                    if(s[i+1] == 'B')
                    {
                        if(i+2<s.size())
                        {

                            s = s.substr(0, i) + s.substr(i+2);
                        }
                        else
                        {
                            s = s.substr(0, i);
                        }
                        removed = true;
                    }
                }
            }

            if(s[i] == 'C')
            {
                if(i+1 < s.size())
                {
                    if(s[i+1] == 'D')
                    {
                        if(i+2<s.size())
                        {

                            s = s.substr(0, i) + s.substr(i+2);
                        }
                        else
                        {
                            s = s.substr(0, i);
                        }
                        removed = true;
                    }
                }
            }
        }
        if(removed)
        {
            return minLength(s);
        }
        else
        {
            return s.size();
        }
    }
};

File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.cs
public class Solution {
    public int MinLength(string s) {
        Stack myStack = new Stack();
        int index = 0;

        while(index < s.Length) 
        {
            myStack.Push(s[index]);
            while((char)myStack.Peek() == 'B' || (char)myStack.Peek() == 'D')
            {
                char temp = (char)myStack.Pop();
                bool removed = false;
                if(myStack.Count != 0)
                {
                    
                    char secondTemp = (char)myStack.Peek();
                    //Console.WriteLine($"{temp} {secondTemp}");
                    if(secondTemp == 'A' && temp == 'B')
                    {
                        myStack.Pop();
                        removed = true;
                    }
                    if(secondTemp == 'C' && temp == 'D')
                    {
                        myStack.Pop();
                        removed = true;
                    }
                }

                if(removed == false)
                {
                    myStack.Push(temp);
                    break;
                }
                if(myStack.Count == 0)
                break;
            }
            index++;
        }
        return myStack.Count;
    }
}